services/web/server/src/simcore_service_webserver/catalog/_api.py

```diff
@@ -56,9 +60,113 @@ def create(cls, request: Request) -> "CatalogRequestContext":
             )
 
 
+async def _safe_replace_service_input_outputs(
+    service: dict[str, Any], unit_registry: UnitRegistry
+):
+    try:
+        await asyncio.to_thread(
+            replace_service_input_outputs,
+            service,
+            unit_registry=unit_registry,
+            **RESPONSE_MODEL_POLICY,
+        )
+    except KeyError:  # noqa: PERF203
+        # This will limit the effect of a any error in the formatting of
+        # service metadata (mostly in label annotations). Otherwise it would
+        # completely break all the listing operation. At this moment,
+        # a limitation on schema's $ref produced an error that made faiing
+        # the full service listing.
+        _logger.exception(
+            "Failed while processing this %s. "
+            "Skipping service from listing. "
+            "TIP: check formatting of docker label annotations for inputs/outputs.",
+            f"{service=}",
+        )
+
+
 # IMPLEMENTATION --------------------------------------------------------------------------------
 
 
+async def dev_list_latest_services(
```

Comment:
One observation here. Usually we call it mock data, if it's mock data I think it should be called mock and not dev.

Reply:
@GitHK there is no mock data here. it is the real call. 
These are prefixed dev_ to distinguish them from the function sthey will replace